pkg/client/zitadel/client.go

```diff
@@ -38,17 +40,27 @@ func NewConnection(issuer, api string, scopes []string, options ...Option) (*Con
 		}
 	}
 
-	err := c.setInterceptors(c.issuer, c.orgID, c.scopes, c.jwtProfileTokenSource)
-	if err != nil {
-		return nil, err
-	}
-	dialOptions := []grpc.DialOption{
-		grpc.WithChainUnaryInterceptor(
-			c.unaryInterceptors...,
-		),
-		grpc.WithChainStreamInterceptor(
-			c.streamInterceptors...,
-		),
+	dialOptions := []grpc.DialOption{}
+	if c.staticTokenSource == "" {
+		err := c.setInterceptors(c.issuer, c.orgID, c.scopes, c.jwtProfileTokenSource)
+		if err != nil {
+			return nil, err
+		}
+		dialOptions = append(dialOptions,
+			grpc.WithChainUnaryInterceptor(
+				c.unaryInterceptors...,
+			),
+			grpc.WithChainStreamInterceptor(
+				c.streamInterceptors...,
+			),
+		)
+	} else {
+		c.setCredentials(c.staticTokenSource)
+		dialOptions = append(dialOptions,
+			grpc.WithPerRPCCredentials(
+				c.perRPCCredentials,
+			),
+		)
```

Comment:
the setInterceptors method also sets an interceptor for the `x-zitadel-orgid` header (allows to specify another organisation if needed)
IMO the static token version might need that as well